lib/api/apiUtils/object/deleteObject.js

```diff
@@ -1,18 +1,22 @@
 /**
  * _bucketRequiresOplogUpdate - DELETE an object from a bucket
- * @param {BucketInfo} bucket - bucket object
+ * @param {ObjectMD} objMD - object metadata
+ * @param {BucketInfo} bucket - bucket info
  * @return {boolean} whether objects require oplog updates on deletion, or not
  */
-function _bucketRequiresOplogUpdate(bucket) {
+function _deleteRequiresOplogUpdate(objMD, bucket) {
+    // If the object is archived, an oplog update is required
+    if (objMD.archive) {
+        return true;
+    }
     // Default behavior is to require an oplog update
-    if (!bucket || !bucket.getLifecycleConfiguration || !bucket.getNotificationConfiguration) {
+    if (!bucket || !bucket.getNotificationConfiguration) {
         return true;
     }
-    // If the bucket has lifecycle configuration or notification configuration
-    // set, we also require an oplog update
-    return bucket.getLifecycleConfiguration() || bucket.getNotificationConfiguration();
+    // If the bucket has notification configuration set, we also require an oplog update
+    return bucket.getNotificationConfiguration();
```

Comment:
Just to satisfy the function's contract of returned type

```suggestion
    return Boolean(bucket.getNotificationConfiguration());
```

We may also use `!= null` but converting to Boolean avoids changing the behavior in case it's `undefined`.